frontend/src/store/questions/index.js

```diff
@@ -51,15 +50,25 @@ const mutations = {
 }
 
 const actions = {
-  async getFeed ({commit}) {
+  async doRequest({commit}, {action, args}) {
     commit('TOGGLE_LOADING', true)
-    commit('REFRESH_FEED', await getQuestions())
-    commit('TOGGLE_LOADING', false)
+    try {
+      const data = await REQUEST[action](...args)
+      commit(COMMITS[action], data)
+    } catch (e) {
+      commit('API_ERROR', e)
+    } finally {
+      commit('TOGGLE_LOADING', false)
+    }
```

Comment:
okay I like this. It's what I was trying to do but better

Reply:
Yeah it's nice ay.